### PR TITLE
Release v5.2.0

### PR DESCRIPTION
**Pull request by tynandebold**
## Description

Release v5.2.0.

It's a minor-version release because we're adding some new functionality as well as some fixes.

> **Note**
> Also contained in this PR is an update to the demo project to use the latest version of Kedro (`0.18.4`).

Here's what's going in:

#### Major features and improvements

- Allow users to hide modular pipelines on the flowchart. (#1046)
- Create URL parameters for each element/section in the flowchart. (#1138)
- Improve CLI loading time. (#1196)
- Make Kedro-Viz compatible with kedro-datasets. (#1214)
- Add time series and parallel coordinates metrics plots to experiment tracking. (#1102)

#### Bug fixes and other changes

- Fix pretty naming for transcoded datasets. (#1062)
- Keep graph visible when switching pipelines. (#1063)
- Add a transition to run details in experiment tracking. (#1065)
- Remove Tags field from dataset and parameter nodes when viewed in metadata panel. (#1100)
- Fix keyboard support for exporting experiment runs. (#1103)
- Fix `MemoryDataSet` not displaying on metadata panel. (#1113)
- Enhance display of the flowchart when hovering over the FocusMode icon. (#1107)
- Make dotted datasets clickable and hoverable when in focus mode. (#1114)
- Fix a bug where tall Matplotlib images weren't displaying correctly. (#1145)

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by AntonyMilneQB — APPROVED** [review]
Looks great, and I think 5.2.0 is right version for this. I've been waiting for this so I could try out the new parallel coordinates plot on the demo website 😀

**Comment by AntonyMilneQB** [comment]
Wait, one thing I think is missing from the release notes. Give me a second and I'll add it.

**Review by AntonyMilneQB — CHANGES_REQUESTED** [review]
...

**Review by Huongg — APPROVED** [review]
LGTM 👍

File: RELEASE.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ Please follow the established format:
 - Enhance display of the flowchart when hovering over the FocusMode icon. (#1107)
 - Make dotted datasets clickable and hoverable when in focus mode. (#1114)
 - Fix a bug where tall Matplotlib images weren't displaying correctly. (#1145)
+- Non-JSON serialisable YAML parameter values (`.nan` and `.inf`) are now rendered as `null` (#1087).
```
Comment (AntonyMilneQB) [inline RELEASE.md:30]:
```suggestion
- Non-JSON serialisable YAML parameter values (`.nan` and `.inf`) are now rendered as `null`. (#1087)

```

**Review by AntonyMilneQB — APPROVED** [review]

**Review by merelcht — APPROVED** [review]
🎉

**Pull request merged**